docs/contributing/pull-requests/index.md

```diff
@@ -0,0 +1,152 @@
+
+```bash
+# Example for the clients repository, from the repository directory
+git remote add upstream https://github.com/bitwarden/clients.git
+```
+
+This will allow you to pull in upstream changes easily by running.
+
+```bash
+# Example for the clients repository, from the repository directory
+git fetch upstream
+```
+
+</community>
+
+## Branch
+
+Each new feature or bug fix should be developed on a separate branch. Branches allows you to work on
+multiple features concurrently. In most cases you should branch from `master`. However, if you are
+working with other contributors we typically branch of a long-lived feature branch. Long-lived
+feature branches allows us to break up a single feature into multiple PRs, which can be reviewed
+individually.
+
+<community>
+
+As a community contributor you can use the following command to branch directly from the _upstream_
+master branch.
+
+```bash
+git checkout -b feature/example
+```
+
+</community>
+
+<bitwarden>
+
+As a Bitwarden contributor you should branch of `origin/master`, this ensures that the branch is
+always based of the latest upstream `master` even if the local `master` is out of date.
+
+```bash
+git checkout -b <pod>/<feature>/<issue-number> -t origin/master
+```
+
+</bitwarden>
+
+<bitwarden>
+
+### Naming convention
+
+To keep branches organized we adhere a naming convention for branches. The convention splits branch
+names into 2 components, each separated by a `/`.
+
+1. The developers pod abbreviation (e.g. `sm`)
+2. Jira issue tag (e.g. `SM-1234`)
+
+```bash
+# Some examples
+sm/SM-1234
+ps/PS-4321-optional-description
+```
+
+This naming convention allows us to easily identify the type of work being done on a branch. And
+will assist in identifying and tracking down stale branches.
+
+</bitwarden>
+
+## Commit
+
+We recommend grouping related changes together into a single commit. This can make it easier for
+reviewers to understand and assess the changes that are being proposed, while also giving the
+contributor checkpoints to revert to if something should go wrong.
+
+More advanced contributors might find it useful to
+[Rewrite History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History). This allows a
+contributor to revise their local history before pushing to the remote repository. A common use case
+is squashing multiple half-working commits. Please be sure to follow the
+[force-pushing recommendations](#force-pushing).
+
+## Creating a Pull Request
+
+The Bitwarden repositories have a _Pull Request template_ which should be followed. This will ensure
+the PR review goes smoothly since it will provide context to the reviewer. <community> Once a
+community PRs has been created, they will be automatically be linked to an internal Jira ticket. The
+internal ticket is used for prioritization and tracking purposes. </community>
+
+<bitwarden>
+
+Please follow the following checklist when creating a PR.
+
+- The PR should always contain a Jira reference.
```

Comment:
Should we also include a reminder to fill out the PR Overview and document the files changed?  I have been stressing this as being important to the new engineers on the team, so I would think it would make sense to call out in the documentation.  If we no longer feel that it's important, we should probably change the templates to reflect that.

Reply:
Ah didn't know we had that documented. Yes we should merge that page into this documentation to lower the number of places we have to look at. Are you alright with having that potentially be a fast follow up PR? Would prefer to avoid scope creeping this and just get it merged.

Reply:
Absolutely.  I can do that if you don't have the time.